deploy/ndm-operator.yaml

```diff
@@ -36,58 +507,93 @@ data:
+        exclude: "/dev/loop,/dev/fd0,/dev/sr0,/dev/ram,/dev/md,/dev/dm-,/dev/rbd,/dev/zd"
 ---
+# Create NDM Service Account
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: openebs-maya-operator
+  namespace: openebs
+---
+# Define Role that allows operations on K8s pods/deployments
+# in "openebs" namespace
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  namespace: openebs
+  name: openebs-ndm-operator
+rules:
+- apiGroups: ["*"]
+  resources: ["nodes", "pods", "services", "endpoints", "events", "configmaps", "secrets", "jobs"]
+  verbs:
+  - '*'
+- apiGroups: ["apiextensions.k8s.io"]
+  resources: ["customresourcedefinitions"]
+  verbs:
+  - '*'
+- apiGroups:
+  - openebs.io
+  resources:
+  - blockdevices
+  - blockdeviceclaims
+  verbs:
+  - '*'
+---
+# Bind the Service Account with the Role Privileges.
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: openebs-ndm-operator
+  namespace: openebs
+subjects:
+- kind: ServiceAccount
+  name: openebs-maya-operator
+  namespace: openebs
+roleRef:
+  kind: ClusterRole
+  name: openebs-ndm-operator
+  apiGroup: rbac.authorization.k8s.io
+---
+# Create NDM daemonset
 apiVersion: apps/v1
 kind: DaemonSet
 metadata:
-  name: openebs-ndm
+  name: node-disk-manager
   namespace: openebs
-  labels:
-    name: openebs-ndm
-    openebs.io/component-name: ndm
-    openebs.io/version: 2.6.0
 spec:
-  selector:
-    matchLabels:
-      name: openebs-ndm
-      openebs.io/component-name: ndm
   updateStrategy:
     type: RollingUpdate
+  selector:
+    matchLabels:
+      name: node-disk-manager
   template:
     metadata:
       labels:
-        name: openebs-ndm
-        openebs.io/component-name: ndm
-        openebs.io/version: 2.6.0
+        name: node-disk-manager
     spec:
       # By default the node-disk-manager will be run on all kubernetes nodes
       # If you would like to limit this to only some nodes, say the nodes
-      # that have storage attached, you could label those node and use
-      # nodeSelector.
-      #
-      # e.g. label the storage nodes with - "openebs.io/nodegroup"="storage-node"
+      # that have storage attached, you could label those node and use nodeSelector.
+      # Example: Label the storage nodes with - "openebs.io/nodegroup"="storage-node"
       # kubectl label node <node-name> "openebs.io/nodegroup"="storage-node"
-      #nodeSelector:
-      #  "openebs.io/nodegroup": "storage-node"
-      serviceAccountName: openebs-maya-operator
+      # nodeSelector:
+      #   "openebs.io/nodegroup": "storage-node"
+      # Use host network as container network to monitor udev source using netlink
+      # to detect disk attach and detach events using fd.
       hostNetwork: true
-      # host PID is used to check status of iSCSI Service when the NDM
-      # API service is enabled
-      #hostPID: true
+      hostPID: true
+      serviceAccountName: openebs-maya-operator
```

Comment:
The service account needs to be changed here.

Reply:
done